src/errors.rs

```diff
@@ -23,7 +23,7 @@ pub enum InternalError {
     CallingApplicationMistake(#[from] CallerError),
     #[error("Serialization Error")]
     Serialization,
```

Comment:
I think this one is context dependent as to whether it's your fault or an external party's and so probably requires a bit of work